Add documentation about media middleware (#34)
front-commerce/developers.front-commerce.com: source/docs/advanced/production-ready/media-middleware.md

---
id: media-middleware
title: Optimize your media
---

> One of the tough things to do when doing responsive websites is to have images that match your user's screen. Especially when content is contributed by a wide range of people that aren't fully aware of the impact their actions can have on the user's experience.

Reviewer: I’d add a first short paragraph summarizing the problem.

Suggestions:

> Heavy, unoptimized images make your site noticeably slow for your users and damage their experience.

Also:

```suggestion
One of the tough things to do when doing responsive websites is to have images that match your user's screen. This is especially true when content is contributed by a wide range of people who aren't fully aware of the impact their actions can have on the user's experience. Iterating fast and last-minute design changes may also lead to introducing heavy and unoptimized images without paying attention.
```

> To solve this issue, Front-Commerce has what we call a media middleware. It is a proxy that will fetch your media from your upload server (Magento, Wordpress, etc.) resize it to the requested size, cache it, and send it back to the user request. This is kind of like the service of [Cloudinary](https://cloudinary.com/documentation/responsive_images).

Reviewer: ```suggestion
To solve this issue, Front-Commerce has what we call a **media middleware**. It is a proxy that will fetch your media from your upload server (Magento, Wordpress, etc.) resize it to the requested size, cache it, and send it back to the user request. This is close to what services such as [Cloudinary](https://cloudinary.com/documentation/responsive_images) provide.
```

<blockquote class="note">
This method has two advantages:

> * you no longer need to expose your backend since it will be the Front-Commerce server that will fetch the image on your backend server

Reviewer: We should use `<ul>`/`<li>` here.
